modules/nw-ipfailover-remove.adoc

```diff
@@ -0,0 +1,144 @@
+// Module included in the following assemblies:
+//
+// * networking/configuring-ipfailover.adoc
+
+[id="nw-ipfailover-remove_{context}"]
+= Removing IP failover
+
+When IP failover is initially configured, the worker nodes in the cluster are modified with an IP tables rule that explicitly allows multicast packets on `224.0.0.18` for Keepalived.
+Because of the change to the nodes, removing IP failover requires running a job to remove the IP tables rule.
+
+.Procedure
+
+. Identify an existing deployment for IP failover:
++
+[source,terminal]
+----
+$ oc get deployment -l ipfailover --all-namespaces
+----
++
+.Example output
+[source,terminal]
+----
+NAMESPACE   NAME         READY   UP-TO-DATE   AVAILABLE   AGE
+default     ipfailover   2/2     2            2           105d
+----
+
+. Delete the deployment:
++
+[source,terminal]
+----
+$ oc delete deployment <ipfailover_deployment_name>
+----
+
+. Optional: Identify and delete any check and notify scripts that are stored as config maps.
+
+.. Identify whether any pods for IP failover use a config map as a volume:
++
+[source,terminal]
+----
+$ oc get pod -l ipfailover -A \
+  -o jsonpath="\
+{range .items[?(@.spec.volumes[*].configMap)]}
+{'Namespace: '}{.metadata.namespace}
+{'Pod:       '}{.metadata.name}
+{'Volumes that use config maps:'}
+{range .spec.volumes[?(@.configMap)]}  {'volume:    '}{.name}
+  {'configMap: '}{.configMap.name}{'\n'}{end}
+{end}"
+----
++
+.Example output
+----
+Namespace: default
+Pod:       keepalived-worker-59df45db9c-2x9mn
+Volumes that use config maps:
+  volume:    config-volume
+  configMap: mycustomcheck
+----
+
+.. If the preceding step provided the names of config maps that are used as volumes, delete the config maps:
++
+[source,terminal]
+----
+$ oc delete configmap -n <namespace> <configmap_name>
+----
+
+. Remove the security context constraints for the `ipfailover` service account.
+
+.. Remove the `privileged` security context constraint:
++
+[source,terminal]
+----
+$ oc adm policy remove-scc-from-user privileged -z ipfailover
+----
+
+.. Remove the `hostNetwork` security context constraint:
++
+[source,terminal]
+----
+$ oc adm policy remove-scc-from-user hostnetwork -z ipfailover
+----
+
```

Comment:
Is it necessary to manually remove the security context constraints? I'm not super knowledgeable in this area, but I'd think sccs would be cleaned up when deleting the service account in the next step.

Reply:
@rfredette, I think you are correct, or at least the key is that the doc specifies `spec.serviceAccountName: ipfailover` and so if the service account is deleted, then customers would need to play by some other rules than what is documented.

FWIW:

```
$ oc adm policy scc-review -z ipfailover -f ipfailover.yaml 
RESOURCE                           SERVICE ACCOUNT   ALLOWED BY   
Deployment/ipfailover-keepalived   ipfailover        privileged   


$ oc delete sa ipfailover
serviceaccount "ipfailover" deleted


$ oc adm policy scc-review -f ipfailover.yaml              
error: unable to compute Pod Security Policy Review for "ipfailover-keepalived": unable to retrieve ServiceAccount ipfailover: serviceaccount "ipfailover" not found

$ oc adm policy scc-review -z ipfailover -f ipfailover.yaml
error: unable to compute Pod Security Policy Review for "ipfailover-keepalived": unable to retrieve ServiceAccount ipfailover: serviceaccount "ipfailover" not found
```

Thank you.